packages/build/src/plugins_core/functions/index.js

```diff
@@ -36,19 +36,31 @@ const normalizeFunctionConfig = ({ buildDir, functionConfig = {} }) => ({
   // fallback behavior ourselves. We do this by transforming the value
   // `esbuild` into `esbuild_zisi`, which zip-it-and-ship-it understands.
   nodeBundler: functionConfig.node_bundler === 'esbuild' ? 'esbuild_zisi' : functionConfig.node_bundler,
+
+  // When the `zisiHandlerV2` feature flag is present, zip-it-and-ship-it will
+  // use the experimental function handler changes introduced in v4.2.0.
+  basePath: featureFlags.zisiHandlerV2 ? buildDir : undefined,
+  experimentalHandlerV2: Boolean(featureFlags.zisiHandlerV2),
```

Comment:
Could we add this flag to all tests? For example, [here](https://github.com/netlify/build/blob/2183f570b44956a4a3d3c1e2e64baccdf7e156b1/packages/build/tests/helpers/main.js#L25). 
Some of the tests run ZISI so that would give another round of automated tests to this feature.

Reply:
(Discussed in Slack. Will update the tests once this behaviour is made the default.)

Reply:
Discussed with @eduardoboucas and we will want to do this later on, before we start the progressive rollout on LaunchDarkly.